flask_admin/model/base.py

```diff
@@ -2040,6 +2045,28 @@ def export_csv(self):
                                     view_args.search, view_args.filters,
                                     page_size=self.export_max_rows)
 
+        return count, data
+
+    @expose('/export/<export_type>/')
+    def export(self, export_type):
+        if export_type == 'csv':
+            return self.export_csv()
+        else:
+            return self._export_tablib(export_type)
+
+    @expose('/export/csv/')
+    def export_csv(self):
+        """
+            Export a CSV of records.
+        """
+        return_url = get_redirect_target() or self.get_url('.index_view')
+
+        if not self.can_export:
```

Comment:
This "permission denied" code is in both `export_csv` and `_export_tablib`, you can consolidate in `export` now.